test/encoder/encoding_test.exs

```diff
@@ -0,0 +1,56 @@
+defmodule DecodingTest do
+  import Membrane.Testing.Assertions
+  alias Membrane.Testing.Pipeline
+  alias Membrane.Element
+  use ExUnit.Case
+
+  def prepare_paths(filename) do
+    in_path = "../fixtures/reference-#{filename}.raw" |> Path.expand(__DIR__)
+    out_path = "/tmp/output-encode-#{filename}.h264"
+    File.rm(out_path)
+    on_exit(fn -> File.rm(out_path) end)
+    {in_path, out_path}
+  end
+
+  def make_pipeline(in_path, out_path, width, height, format \\ :I420) do
+    Pipeline.start_link(%Pipeline.Options{
+      elements: [
+        file_src: %Element.File.Source{chunk_size: 40_960, location: in_path},
+        parser: %Element.RawVideo.Parser{width: width, height: height, format: format},
+        encoder: %Element.FFmpeg.H264.Encoder{preset: :fast, crf: 30},
+        sink: %Element.File.Sink{location: out_path}
+      ],
+      links: %{
+        {:file_src, :output} => {:parser, :input},
+        {:parser, :output} => {:encoder, :input},
+        {:encoder, :output} => {:sink, :input}
+      }
+    })
+  end
+
+  describe "EncodingPipeline should" do
```

Comment:
All of tests in this section are identical, only path changes. Maybe their contents could be replaced by function definition and function call in each test body with file name as parameter?